edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +224,32 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let lock = if let ConcurrencyThrottling::Limited(limit) = concurrency {
+            Some((Arc::new(AtomicU32::new(limit)), limit))
+        } else {
+            None
+        };
+
         let srv = incoming.for_each(move |(socket, addr)| {
-            let protocol = protocol.clone();
 
+            if let Some((lock, limit)) = lock.clone() {
+                let limit_reached = lock.fetch_update( Ordering::AcqRel,
+                    Ordering::Acquire,
```

Comment:
@arsing maybe you know more on this, but does AcqRel do anything here? I believe the first ordering is for the set and the second for the load.

Reply:
Yes, that's correct. I do not understand it fully though. I think the "set" is when you take the lock and increment. While the second is just fetching the data.

Reply:
Yeah, I am wondering if it can just be Release? As the second one is Acquire... but I am not sure...

Reply:
You don't have to wonder. :) 

>https://doc.rust-lang.org/stable/std/sync/atomic/struct.AtomicUsize.html#method.fetch_update

>`fetch_update` takes two `Ordering` arguments to describe the memory ordering of this operation. The first describes the required ordering for when the operation finally succeeds while the second describes the required ordering for loads. [...]

For the operation to "finally succeed", it has to fetch *and* update, so both read-and-write. And even if you don't believe that...

>[...] These correspond to the success and failure orderings of `AtomicUsize::compare_exchange` respectively.

-->

>https://doc.rust-lang.org/stable/std/sync/atomic/struct.AtomicUsize.html#method.compare_exchange

>`compare_exchange` takes two `Ordering` arguments to describe the memory ordering of this operation. `success` describes the required ordering for the read-modify-write operation that takes place if the comparison with `current` succeeds.

**read-modify-write**

---

(In fact, `(Release, Acquire)` will straight-up panic: https://doc.rust-lang.org/stable/src/core/sync/atomic.rs.html#2449-2462 )